### PR TITLE
nsqlookupd: improve test coverage

### PR DESCRIPTION
**Pull request by kenjones-cisco**
Adds test cases to improve code coverage to protect against potential
compatibility conflicts moving forward.

Existing test cases refactored to leverage common test helpers within
the internal/test package to avoid code duplication and simplify
the testing effort.

### CODE REVIEW

**Comment by kenjones-cisco** [comment]
Initial efforts, so far but wanted to start the PR so that it is visible and if there is any feedback on the work being done that it can be shared sooner rather than later.

I'll remove the [WIP] from the title once I think it is ready for merging. All feedback welcome!

File: internal/test/assertions.go [hunk 1]
```diff
@@ -17,10 +17,10 @@ func Equal(t *testing.T, expected, actual interface{}) {
 }
 
 func NotEqual(t *testing.T, expected, actual interface{}) {
-	if !reflect.DeepEqual(expected, actual) {
+	if reflect.DeepEqual(expected, actual) {
```
Comment (mreiferson) [inline internal/test/assertions.go]: hah, guess we didn't use that very much 😉

File: nsqlookupd/options.go [hunk 2]
```diff
@@ -16,7 +18,7 @@ type Options struct {
 	InactiveProducerTimeout time.Duration `flag:"inactive-producer-timeout"`
 	TombstoneLifetime       time.Duration `flag:"tombstone-lifetime"`
 
-	Logger logger
+	Logger app.Logger
```
Comment (mreiferson) [inline nsqlookupd/options.go]: `nsqlookupd` is technically a public package and `app` is now an `internal` one - seems like that might be undesirable?

**Comment by mreiferson** [comment]
@kenjones-cisco thanks, this is really great 😍 ❤️ 💯

File: nsqlookupd/options.go (hunk 2 above)
Comment (kenjones-cisco) [inline nsqlookupd/options.go]: Any other idea for providing the common types w/o duplicating the same within each of the public packages?

Comment (mreiferson) [inline nsqlookupd/options.go]: It's such a small interface, I'm ok with a little copy/paste.

**Comment by kenjones-cisco** [comment]
ok, `logger.go` restored and the use of `internal/app/logger.go` has been removed from `nsqlookupd`

**Comment by mreiferson** [comment]
@kenjones-cisco thanks, we've been trying to slowly remove our use of `go-simplejson` throughout the codebase (finished in #367).

It's my fault for not mentioning this earlier on this PR, but would you mind removing `go-simplejson` here so we don't add more back into the code base? I would understand if you're not up for refactoring this, I can handle it in a rebase of #367 if necessary.

**Comment by kenjones-cisco** [comment]
Sure no problem.

**Comment by kenjones-cisco** [comment]
Removed `go-simplejson` from `nsqlookupd` tests.

**Comment by mreiferson** [comment]
😍👍🏻

**Pull request merged**